components/protected.js

```diff
@@ -0,0 +1,27 @@
+import { useAuth0 } from "@auth0/auth0-react";
+import { useRouter } from "next/router";
+
+//check if you are on the client (browser) or server
+const isBrowser = () => typeof window !== "undefined";
+
+const ProtectedRoute = ({ children }) => {
```

Comment:
This doesn't get used anywhere. We are going to need something like this for displaying some features. (should probably have a nav on public pages so users can easily get to their feed.

Reply:
Maybe? OF doesn't do this, I wonder if it's a legal reason. (Infact they display a basic html page.